File: mobile/src/main/java/at/shockbytes/remote/network/model/text/RemiKeyEvent.java
package at.shockbytes.remote.network.model.text;

/**
 * @author Martin Macheiner
 *         Date: 10.10.2017.
 */

public abstract class RemiKeyEvent {

    private int keyCode;
    private String display;

    public RemiKeyEvent(String display, int keyCode) {
        this.display = display;
        this.keyCode = keyCode;
    }

    public int getKeyCode() {
        return keyCode;
    }

    public String getDisplayString() {
        return display;
    }

    public abstract int layout();

    public abstract int getRowSpan();

    public abstract boolean capsLockSensitive();

    @Override
    public boolean equals(Object obj) {

        if (obj instanceof RemiKeyEvent) {
            RemiKeyEvent o = (RemiKeyEvent) obj;
            return keyCode == o.keyCode;
        }
        return super.equals(obj);
    }
}

File: mobile/src/main/java/at/shockbytes/remote/network/model/text/RemiKeyEvent.kt
package at.shockbytes.remote.network.model.text

/**
 * @author Martin Macheiner
 * Date: 10.10.2017.
 */

abstract class RemiKeyEvent constructor(val displayString: String, val keyCode: Int) {

    abstract fun layout(): Int

    abstract fun capsLockSensitive(): Boolean

    abstract fun rowSpan() : Int

    override fun equals(other: Any?): Boolean {

        if (other is RemiKeyEvent) {
            val o = other as RemiKeyEvent?
            return keyCode == o!!.keyCode
        }
        return super.equals(other)
    }

    override fun hashCode(): Int {
        var result = displayString.hashCode()
        result = 31 * result + keyCode
        return result
    }
}
